src/app/api/job/[jobId]/results/route.ts

```diff
@@ -13,50 +13,46 @@ export const POST = wrapApiOrgAction(async (req: Request, { params }: { params:
         return new NextResponse('Unauthorized', { status: 401 })
     }
 
-    try {
-        const formData = await req.formData()
-        const contents = formData.get('file') as File
-
-        if (!contents) {
-            return NextResponse.json({ status: 'fail', error: 'no "file" entry in post data' }, { status: 500 })
-        }
-
-        // join is a security check to ensure the job is owned by the org
-        const info = await db
-            .selectFrom('studyJob')
-            .innerJoin('study', (join) =>
-                join.onRef('study.id', '=', 'studyJob.studyId').on('study.orgId', '=', org.id),
-            )
-            .select(['studyJob.id as studyJobId', 'studyId'])
-            .where('studyJob.id', '=', jobId)
-            .executeTakeFirst()
-
-        if (!info) {
-            return NextResponse.json({ status: 'fail', error: 'job not found' }, { status: 404 })
-        }
-
-        await storeStudyResultsFile(
-            {
-                ...info,
-                orgSlug: org.slug,
-                resultsType: 'ENCRYPTED',
-            },
-            contents,
-        )
-
-        await db
-            .insertInto('jobStatusChange')
-            .values({
-                status: 'RUN-COMPLETE',
-                studyJobId: info.studyJobId,
-            })
-            .execute()
-
-        await sendResultsReadyForReviewEmail(info.studyId)
-
-        return NextResponse.json({ status: 'success' }, { status: 200 })
-    } catch (e) {
-        console.error(e)
-        return NextResponse.json({ status: 'fail', error: e }, { status: 500 })
+    const formData = await req.formData()
+    const logs = formData.get('log')
+    let results = formData.get('result')
+    const file = formData.get('file')
+
+    // TODO: remove this once TOA no longer sends 'file' property
+    if (!results && file) {
+        results = file
+    }
+
+    // join is a security check to ensure the job is owned by the org
+    const info = await db
+        .selectFrom('studyJob')
+        .innerJoin('study', (join) => join.onRef('study.id', '=', 'studyJob.studyId').on('study.orgId', '=', org.id))
+        .innerJoin('org', 'org.id', 'study.orgId')
+        .select(['studyJob.id as studyJobId', 'studyId', 'org.slug as orgSlug', 'studyJob.id as studyJobId'])
+        .where('studyJob.id', '=', jobId)
+        .executeTakeFirst()
+
+    if (!info) {
+        return NextResponse.json({ status: 'fail', error: 'job not found' }, { status: 404 })
+    }
+
+    if (logs instanceof File) {
+        await storeStudyEncryptedLogFile(info, logs)
     }
+
+    if (results instanceof File) {
+        await storeStudyEncryptedResultsFile(info, results)
+    }
+
+    await db
+        .insertInto('jobStatusChange')
+        .values({
+            status: logs && !results ? 'JOB-ERRORED' : 'RUN-COMPLETE', // TODO: verify this is correct status,
```

Comment:
FYI setting the `JOB-ERRORED` status here will duplicate the Setup App doing the same thing https://github.com/safeinsights/setup-app/blob/main/src/lib/aws-check-jobs.ts#L55